src/core/Mixer.cpp

```diff
@@ -318,10 +319,13 @@ const surroundSampleFrame * Mixer::renderNextBuffer()
 
 	static Song::PlayPos last_metro_pos = -1;
 
-	Song::PlayPos p = Engine::getSong()->getPlayPos(
-						Song::Mode_PlayPattern );
-	if( Engine::getSong()->playMode() == Song::Mode_PlayPattern &&
-		gui->pianoRoll()->isRecording() == true &&
+	Song::PlayModes currentPlayMode = Engine::getSong()->playMode();
+	Song::PlayPos p = Engine::getSong()->getPlayPos( currentPlayMode );
+
+	if( (   currentPlayMode == Song::Mode_PlayPattern || currentPlayMode == Song::Mode_PlaySong
+	     || currentPlayMode == Song::Mode_PlayBB ) &&
+		m_metronomeActive &&
+		!Engine::getSong()->isExporting() &&
 		p != last_metro_pos )
 	{
```

Comment:
What do you think of decomposing the conditional a bit? Maybe something like:
```c++
bool isPlaying = currentPlayMode == Song::Mode_PlayPattern || currentPlayMode == Song::Mode_PlaySong
	     || currentPlayMode == Song::Mode_PlayBB;
if (isPlaying && m_metronomeActive && !Engine::getSong()->isExporting() &&
 		p != last_metro_pos) ...
```

Hope I'm not being too pedantic. Everything else looks good.